addons/account/models/account_analytic_distribution_model.py

```diff
@@ -32,11 +33,34 @@ def _get_default_search_domain_vals(self):
 
     def _get_applicable_models(self, vals):
         applicable_models = super()._get_applicable_models(vals)
-        return applicable_models.filtered(lambda m:
-            not m.account_prefix or vals.get('account_prefix', '').startswith(m.account_prefix)
+
+        # Regex pattern to split by either ';' or ','
+        delimiter_pattern = re.compile(r'[;,]\s*')
+
+        return applicable_models.filtered(lambda model:
+            not model.account_prefix or
```

Comment:
PEP would complain less if you put your lambda model on the following line (but you see what you prefer)